### PR TITLE
docs: add docs of workflow plugin

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes # (issue)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/en/latest/plugins/workflow.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
+---
+title: workflow
+keywords:
+  - APISIX
+  - Plugin
+  - workflow
```
Comment (SylviaBABY) [inline docs/en/latest/plugins/workflow.md:6]:
```suggestion
  - workflow
  - traffic control
```

File: docs/en/latest/plugins/workflow.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+description: This document contains information about the Apache APISIX workflow Plugin.
```
Comment (SylviaBABY) [inline docs/en/latest/plugins/workflow.md:7]:
```suggestion
description: This document describes the Apache APISIX workflow Plugin, you can use it to control traffic.
```

File: docs/zh/latest/plugins/workflow.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+## 描述
+
+`workflow` 插件引入 [lua-resty-expr](https://github.com/api7/lua-resty-expr#operator-list) 来提供复杂的流量控制功能。
+
+## 属性
+
+| 名称          | 类型   | 必选项  | 默认值                    | 有效值                                                                                                                                            | 描述 |
+| ------------- | ------ | ------ | ------------------------ | -------------------------------------------------------------------------------------------------------------------------------------------------- | ------------- |
+| rules.case | array[array] | 是 |  |                                                                                                                            | 由一个或多个{var, operator, val}元素组成的列表，类似这样：{{var, operator, val}, {var, operator, val}, ...}}。例如：{"arg_name", "==", "json"}，表示当前请求参数 name 是 json。这里的 var 与 Nginx 内部自身变量命名是保持一致，所以也可以使用 request_uri、host 等；对于 operator 部分，目前已支持的运算符有 ==、~=、~~、>、<、in、has 和 ! 。操作符的具体用法请看 [lua-resty-expr](https://github.com/api7/lua-resty-expr#operator-list) 的 `operator-list` 部分。 |
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/workflow.md:37]:
```suggestion
| rules.case | array[array] | 是 |  |                                                                                                                            | 由一个或多个 {var, operator, val} 元素组成的列表，类似这样：{{var, operator, val}, {var, operator, val}, ...}}。例如：{"arg_name", "==", "json"}，表示当前请求参数 name 是 json。这里的 var 与 NGINX 内部自身变量命名保持一致，所以也可以使用 request_uri、host 等；对于 operator 部分，目前已支持的运算符有 ==、~=、~~、>、<、in、has 和 ! 。关于操作符的具体用法请参考 [lua-resty-expr](https://github.com/api7/lua-resty-expr#operator-list) 的 `operator-list` 部分。 |
```

File: docs/zh/latest/plugins/workflow.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        "nodes":{
+            "127.0.0.1:1980":1
+        }
+    }
+}'
+```
+
+如上，我们启用了 `workflow` 插件，如果请求与 `rules` 中的 `case` 匹配，则会执行对应的 `actions`。
+
+示例 1: 如果请求的 uri 是 `/hello/rejected`，则返回给客户端状态码 403。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/workflow.md:101]:
```suggestion
**示例 1: 如果请求的 uri 是 `/hello/rejected`，则返回给客户端状态码 `403`**
```

File: docs/zh/latest/plugins/workflow.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+```shell
+curl http://127.0.0.1:9080/hello/rejected -i
+HTTP/1.1 403 Forbidden
+......
+
+{"error_msg":"rejected by workflow"}
+```
+
+示例  2: 如果请求的 uri 是 `/hello/v2/appid`，则执行 `limit-count` 插件，限制请求的数量为 2，时间窗口为 60 秒，如果超过限制数量，则返回给客户端状态码 429
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/workflow.md:111]:
```suggestion
**示例 2: 如果请求的 uri 是 `/hello/v2/appid`，则执行 `limit-count` 插件，限制请求的数量为 2，时间窗口为 60 秒，如果超过限制数量，则返回给客户端状态码 `429`**
```

File: docs/zh/latest/plugins/workflow.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+curl http://127.0.0.1:0080/hello/v2/appid -i
+HTTP/1.1 200 OK
+```
+
+```shell
+curl http://127.0.0.1:0080/hello/v2/appid -i
+HTTP/1.1 429 Too Many Requests
+```
+
+示例 3: 如果请求不能被任何 `case` 匹配，则 `workflow` 不会执行任何操作
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/workflow.md:128]:
```suggestion
**示例 3: 如果请求不能被任何 `case` 匹配，则 `workflow` 不会执行任何操作**
```

File: docs/zh/latest/plugins/workflow.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+curl http://127.0.0.1:0080/hello/fake -i
+HTTP/1.1 200 OK
+```
+
+## Disable Plugin
+
+当你需要禁用 `workflow` 插件时，可以通过以下命令删除相应的 JSON 配置，APISIX 将会自动重新加载相关配置，无需重启服务：
+
+```shell
+curl http://127.0.0.1:9180/apisix/admin/routes/1  -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/workflow.md:140]:
```suggestion
curl http://127.0.0.1:9180/apisix/admin/routes/1 \
-H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
```

File: docs/zh/latest/plugins/workflow.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+在 `rules` 中，按照 `rules` 的数组下标顺序依次匹配 `case`，如果 `case` 匹配成功，则直接执行对应的 `actions`。
+
+:::
+
+## 启用插件
+
+以下示例展示了如何在路由中启用 `workflow` 插件：
+
+```shell
+curl http://127.0.0.1:9180/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/workflow.md:55]:
```suggestion
curl http://127.0.0.1:9180/apisix/admin/routes/1 \
-H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
```

File: docs/zh/latest/plugins/workflow.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
+---
+title: workflow
+keywords:
+  - APISIX
+  - Plugin
+  - workflow
+description: 本文介绍了关于 Apache APISIX `workflow` 插件的基本信息及使用方法。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/workflow.md:7]:
```suggestion
description: 本文介绍了关于 Apache APISIX `workflow` 插件的基本信息及使用方法，你可以基于此插件进行复杂的流量操作。
```

File: docs/en/latest/plugins/workflow.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+curl http://127.0.0.1:0080/hello/fake -i
+HTTP/1.1 200 OK
+```
+
+## Disable Plugin
+
+To disable the `workflow` plugin, you can delete the corresponding JSON configuration from the Plugin configuration. APISIX will automatically reload and you do not have to restart for this to take effect.
+
+```shell
+curl http://127.0.0.1:9180/apisix/admin/routes/1  -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
```
Comment (SylviaBABY) [inline docs/en/latest/plugins/workflow.md:140]:
```suggestion
curl http://127.0.0.1:9180/apisix/admin/routes/1 \
-H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
```

File: docs/en/latest/plugins/workflow.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+curl http://127.0.0.1:0080/hello/v2/appid -i
+HTTP/1.1 200 OK
+```
+
+```shell
+curl http://127.0.0.1:0080/hello/v2/appid -i
+HTTP/1.1 429 Too Many Requests
+```
+
+Example 3: if the request can not match any `case` in the `rules`, the `workflow` plugin would do nothing
```
Comment (SylviaBABY) [inline docs/en/latest/plugins/workflow.md:128]:
```suggestion
**Example 3: if the request can not match any `case` in the `rules`, the `workflow` plugin would do nothing**
```

File: docs/en/latest/plugins/workflow.md [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+
+```shell
+curl http://127.0.0.1:9080/hello/rejected -i
+HTTP/1.1 403 Forbidden
+......
+
+{"error_msg":"rejected by workflow"}
+```
+
+Example 2: if the request uri is `/hello/v2/appid`, the `workflow` plugin would execute the `limit-count` plugin
```
Comment (SylviaBABY) [inline docs/en/latest/plugins/workflow.md:111]:
```suggestion
**Example 2: if the request uri is `/hello/v2/appid`, the `workflow` plugin would execute the `limit-count` plugin**
```

File: docs/en/latest/plugins/workflow.md [hunk 13]
```diff
@@ -0,0 +1,150 @@
...
+        "nodes":{
+            "127.0.0.1:1980":1
+        }
+    }
+}'
+```
+
+Here, the `workflow` Plugin is enabled on the Route. If the request matches the `case` in the `rules`, the `actions` will be executed.
+
+Example 1: if the request uri is `/hello/rejected`, the `workflow` plugin would return directly to the client
```
Comment (SylviaBABY) [inline docs/en/latest/plugins/workflow.md:101]:
```suggestion
**Example 1: If the requested uri is `/hello/rejected`, the status code `403` is returned to the client**
```

File: docs/en/latest/plugins/workflow.md [hunk 14]
```diff
@@ -0,0 +1,150 @@
...
+In `rules`, match `case` in order according to the index of the `rules`, and execute `actions` directly if `case` match.
+
+:::
+
+## Enabling the Plugin
+
+You can configure the `workflow` plugin on a Route as shown below:
+
+```shell
+curl http://127.0.0.1:9180/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
```
Comment (SylviaBABY) [inline docs/en/latest/plugins/workflow.md:55]:
```suggestion
curl http://127.0.0.1:9180/apisix/admin/routes/1 \
-H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
```

File: docs/en/latest/plugins/workflow.md [hunk 15]
```diff
@@ -0,0 +1,150 @@
...
+
+## Description
+
+The `workflow` plugin is used to introduce [lua-resty-expr](https://github.com/api7/lua-resty-expr#operator-list) to provide complex traffic control features.
+
+## Attributes
+
+| Name                         | Type          | Required | Default | Valid values | Description                                                  |
+| ---------------------------- | ------------- | -------- | ------- | ------------ | ------------------------------------------------------------ |
+| rules.case                   | array[array]  | True     |         |              | List of variables to match for filtering requests for conditional traffic split. It is in the format `{variable operator value}`. For example, `{"arg_name", "==", "json"}`. The variables here are consistent with Nginx internal variables. For details on supported operators, [lua-resty-expr](https://github.com/api7/lua-resty-expr#operator-list). |
```
Comment (SylviaBABY) [inline docs/en/latest/plugins/workflow.md:37]:
```suggestion
| rules.case                   | array[array]  | True     |         |              | List of variables to match for filtering requests for conditional traffic split. It is in the format `{variable operator value}`. For example, `{"arg_name", "==", "json"}`. The variables here are consistent with NGINX internal variables. For details on supported operators, you can refer to [lua-resty-expr](https://github.com/api7/lua-resty-expr#operator-list). |
```

File: docs/zh/latest/plugins/workflow.md [hunk 16]
```diff
@@ -0,0 +1,150 @@
+---
+title: workflow
+keywords:
+  - APISIX
+  - Plugin
+  - workflow
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/workflow.md:6]:
```suggestion
  - workflow
  - 流量控制
```

File: docs/en/latest/plugins/workflow.md (hunk 1 above)
Comment (tzssangglass) [reply docs/en/latest/plugins/workflow.md:6]: updated, others too.

**Review by SylviaBABY — DISMISSED** [review]

File: docs/en/latest/config.json [hunk 17]
```diff
@@ -106,7 +106,8 @@
             "plugins/traffic-split",
             "plugins/request-id",
             "plugins/proxy-control",
-            "plugins/client-control"
+            "plugins/client-control",
+            "plugins/workflow"
```
Comment (spacewander) [inline docs/en/latest/config.json:110]: We should put this in the same category of serverless-pre/post plugin.

File: docs/en/latest/plugins/workflow.md [hunk 18]
```diff
@@ -0,0 +1,153 @@
...
+
+| Name                         | Type          | Required | Default | Valid values | Description                                                  |
+| ---------------------------- | ------------- | -------- | ------- | ------------ | ------------------------------------------------------------ |
+| rules.case                   | array[array]  | True     |         |              | List of variables to match for filtering requests for conditional traffic split. It is in the format `{variable operator value}`. For example, `{"arg_name", "==", "json"}`. The variables here are consistent with NGINX internal variables. For details on supported operators, you can refer to [lua-resty-expr](https://github.com/api7/lua-resty-expr#operator-list). |
+| rules.actions                | array[object] | True     |         |              | The action to be performed when the case matches successfully. Currently, only one element is supported in actions. The first child element of the actions' only element can be `return` or `limit-count`. |
+| rules.actions[1].return      | string        | False    |         |              | Return directly to the client.                               |
```
Comment (spacewander) [inline docs/en/latest/plugins/workflow.md:40]: Could we move each action to a separate table? As the number of actions may grow fastly.

File: docs/en/latest/config.json (hunk 17 above)
Comment (tzssangglass) [reply docs/en/latest/config.json]: done

File: docs/en/latest/plugins/workflow.md (hunk 18 above)
Comment (tzssangglass) [reply docs/en/latest/plugins/workflow.md]: updated

**Review by SylviaBABY — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**